### ISSUE TITLE
Node with id=0 cannot be deleted

### ISSUE THREAD

**Issue by FilipQL** [almende/vis#2528]
Hi, 

I have a very simple app created with Vis and the problem is that the node with id=0 cannot be deleted. All of the other nodes can be deleted. 

Here is the example: https://jsfiddle.net/Filip_Z/9k2s3dgj/7/

```
    // create an array with nodes
    var nodes = new vis.DataSet([
        { id: 0, label: '0' },
        { id: 1, label: '1' },
        { id: 2, label: '2' },
        { id: 3, label: '3' },
        { id: 4, label: '4' }
    ]);

    // create an array with edges
    var edges = [];

    // create a network
    var container = document.getElementById('mynetwork');

    // provide the data in the vis format:
    var data = {
        nodes: nodes,
        edges: edges
    };

    var options = {
        manipulation: {
            enabled: true,
            initiallyActive: true,
            deleteNode: function(deleteData, callback) {
                callback(deleteData)
            }
        }
    };

    // initialize your network!
    var network = new vis.Network(container, data, options);
```

As @danieldiekmeier [suggested](https://github.com/vuejs/vue/issues/4622#issuecomment-269956193), this is most likely because Vis checks for truthy-ness in the ids (because if you use a string as the id, it works, and you can remove the node):
```
nodes: [
    { id: '0', label: '0' },
    { id: 1, label: '1' },
    { id: 2, label: '2' },
    { id: 3, label: '3' },
    { id: 4, label: '4' }
]
```

**Comment by lewisjb** [comment]
This appears to actually be a problem with DataSet. I noticed the same problem in some of the Timeline examples ([Ordering of groups](http://visjs.org/examples/timeline/groups/groupsOrdering.html) and [Subgroups](http://visjs.org/examples/timeline/groups/subgroups.html))

I did some debugging, and found that it _is_ because of a 'truthy-ness' check.
The issue is here: [DataSet.js:682](https://github.com/almende/vis/blob/develop/lib/DataSet.js#L682)
Lines:
```javascript
itemId = item[this._fieldId];
if (itemId) {
  removedIds.push(itemId);
  removedItems.push(item);
}
```
I did a manual override in debug mode, to run those two lines for when `id` is 0, and it worked.

It appears the original purpose of the `if` was to ensure that `itemId` existed, in which case changing it to: `if (itemId != undefined)` should resolve the issue without any side-effects.

**Comment by yotamberk** [comment]
fixed with #2530
